test/controllers/update/update.trusts.submission.interrupt.controller.spec.ts

```diff
@@ -0,0 +1,86 @@
+import { getApplicationData } from "../../../src/utils/application.data";
+import { isActiveFeature } from "../../../src/utils/feature.flag";
+
+import { APPLICATION_DATA_MOCK } from '../../__mocks__/session.mock';
+import { PAGE_TITLE_ERROR } from "../../__mocks__/text.mock";
+import { saveAndContinueButtonText } from '../../__mocks__/save.and.continue.mock';
+
+const mockGetApplicationData = getApplicationData as jest.Mock;
+mockGetApplicationData.mockReturnValue( APPLICATION_DATA_MOCK );
+
+const mockAuthenticationMiddleware = authentication as jest.Mock;
+mockAuthenticationMiddleware.mockImplementation((req: Request, res: Response, next: NextFunction) => next() );
+
+const mockCompanyAuthenticationMiddleware = companyAuthentication as jest.Mock;
+mockCompanyAuthenticationMiddleware.mockImplementation((req: Request, res: Response, next: NextFunction) => next() );
+
+const mockServiceAvailabilityMiddleware = serviceAvailabilityMiddleware as jest.Mock;
+mockServiceAvailabilityMiddleware.mockImplementation((req: Request, res: Response, next: NextFunction) => next() );
+
+const mockIsActiveFeature = isActiveFeature as jest.Mock;
+mockIsActiveFeature.mockReturnValue(true);
+
+describe("Update - Trusts - Submission Interrupt", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe("GET tests", () => {
+    test('when feature flag is on, page is returned', async () => {
+      mockIsActiveFeature.mockReturnValue(true);
+
+      const resp = await request(app).get(config.UPDATE_TRUSTS_SUBMISSION_INTERRUPT_URL);
+
+      expect(resp.status).toEqual(200);
+      expect(resp.text).toContain("You now need to submit trust information");
+      expect(resp.text).toContain(config.UPDATE_BENEFICIAL_OWNER_TYPE_URL);
+      expect(resp.text).toContain(saveAndContinueButtonText);
+      expect(resp.text).not.toContain(PAGE_TITLE_ERROR);
+    });
+
+    test('when feature flag is off, 404 is returned', async () => {
+      mockIsActiveFeature.mockReturnValue(false);
+
+      const resp = await request(app).get(config.UPDATE_TRUSTS_SUBMISSION_INTERRUPT_URL);
+
+      expect(resp.status).toEqual(404);
+      expect(resp.text).toContain("Page not found");
+    });
+  });
+
+  describe("POST tests", () => {
+    test('when feature flag is on, redirect to tell us about the trust page', async () => {
+      mockIsActiveFeature.mockReturnValue(true);
+
+      const resp = await request(app).post(config.UPDATE_TRUSTS_SUBMISSION_INTERRUPT_URL);
+
+      expect(resp.status).toEqual(302);
+      expect(resp.header.location).toEqual(config.UPDATE_TRUSTS_TELL_US_ABOUT_IT_URL);
+    });
+
+    test('when feature flag is off, 404 is returned', async () => {
+      mockIsActiveFeature.mockReturnValue(false);
+
+      const resp = await request(app).post(config.UPDATE_TRUSTS_SUBMISSION_INTERRUPT_URL);
+
+      expect(resp.status).toEqual(404);
+      expect(resp.text).toContain("Page not found");
```

Comment:
minor - but there's a PAGE_NOT_FOUND_TEXT variable

Reply:
Now using it